### PR TITLE
CXC-28: Update CARMA Cloud Plugin to have configurable IP/Port

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
Update CARMA Cloud plugin to allow for configurable target IP/Port for sending CARMA Cloud REST Requests
image
image

**NOTE**: Large changeset is due to correcting line ending characters for file. Feel free to double check change set in text diff tool of your choice. Here is what I got
image
image
image
image
image

## Related Issue
[CXC-28](https://usdot-carma.atlassian.net/browse/CXC-28)

## Motivation and Context
To support CARMA Cloud CDASim Integration testing

## How Has This Been Tested?
Locally integration tested using devcontainer to ensure configurable parameters were available and modified URL correctly

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/v2i-hub/CARMACloudPlugin/manifest.json [hunk 1]
```diff
@@ -28,6 +28,21 @@
 			"default":"127.0.0.1",
 			"description":"Server IP address for V2X hub"
 		},
+		{
+			"key":"WebServicePort",
```
Comment (kjrush) [inline src/v2i-hub/CARMACloudPlugin/manifest.json:32]: This name is a bit generic if this port is infact only for TCMs. Maybe rename it something more specific to reflect it's usage?

File: src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp [hunk 2]
```diff
@@ -1,653 +1,657 @@
...
+	GetConfigValue<uint16_t>("TCMRepeatedlyBroadcastTimeOut",_TCMRepeatedlyBroadcastTimeOut);	
+	GetConfigValue<string>("TCMNOAcknowledgementDescription", _TCMNOAcknowledgementDescription);
+	GetConfigValue<int>("TCMRepeatedlyBroadCastTotalTimes", _TCMRepeatedlyBroadCastTotalTimes);
+	GetConfigValue<int>("TCMRepeatedlyBroadcastSleep", _TCMRepeatedlyBroadcastSleep);
+	GetConfigValue<string>("listTCM",list_tcm);
+	std::string carma_cloud_ip;
+	uint carma_cloud_port;
+	GetConfigValue<string>("CARMACloudIP",carma_cloud_ip);
+	GetConfigValue<uint>("CARMACloudPort",carma_cloud_port);
+	carma_cloud_url = carma_cloud_ip + ":" + std::to_string(carma_cloud_port);
```
Comment (kjrush) [inline src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp:502]: Is there a benefit to having these as separate parameters if we're going to concatenate them anyway? It seems like all it's doing is saving the user from having to know that ":" goes between the domain and the port. Would be simpler I think to just have a "carma_cloud_url" parameter in general?

**Review by dan-du-car — APPROVED** [review]

File: src/v2i-hub/CARMACloudPlugin/manifest.json (hunk 1 above)
Comment (paulbourelly999) [reply src/v2i-hub/CARMACloudPlugin/manifest.json:32]: I can rename it but this is not the newly created configuration parameter for carma cloud port and ip. It is the existing one for the IP and port to deploy the CARMA-CLoud Plugin REST Server. Do you have any suggestions for better naming convention. V2X-Hub CARMA Cloud REST Server Port?

Comment (kjrush) [reply src/v2i-hub/CARMACloudPlugin/manifest.json:32]: Ah, if this is pre-existing I'm fine to leave it for now. But something like that, would be preferable. "Web Service Port" just leads me to ask "What web service"? and will likely get slightly more confusing when the new UI framework stuff is finalized, as there may be multiple web services running.

**Pull request merged**